experimental/listener.go

```diff
@@ -24,41 +24,32 @@ type FunctionListener interface {
 	// The returned context will be used as the context of this function call. To add context
 	// information for this function call, add it to ctx and return the updated context. If
 	// no context information is needed, return ctx as is.
-	Before(ctx context.Context) context.Context
+	Before(context.Context, []uint64) context.Context
```

Comment:
probably should label this params and the other results. This may break in later work, but again.. it is experimental, so that's ok!